core/src/main/scala/com/raphtory/RaphtoryApp.scala

```diff
@@ -0,0 +1,38 @@
+package com.raphtory
+
+import cats.effect._
+import cats.effect.unsafe.implicits.global
+import com.raphtory.internals.components.RaphtoryServiceBuilder
+import com.raphtory.internals.context.RaphtoryContext
+import com.raphtory.internals.management.GraphConfig.ConfigBuilder
+
+trait RaphtoryApp {
+
+  sealed trait RaphtoryContextType
```

Comment:
Instead of using this to differentiate the two cases, we could have an object RaphtoryApp containing two traits: Local and Remote. This way we would write MyApp extends RaphtoryApp.Local. Only for the remote one, we could provide methods to override the address. This might be address in a future PR though.

Reply:
I didn't like ContextTypes anyway and this approach seems like a natural fit for our use case since we are anyway allowing only one context per app. Already implemented in this PR itself.